### PR TITLE
Fee Data SubType Refactoring

### PR DESCRIPTION
**Pull request by georgiyazovaliiski**
**Related issue(s)**:
Closes #NONE

**Summary of the change**:
Fee Data refactoring to support SubTypes. Unit Tests were also modified accordingly

**External impacts**:
None.

**Applicable documentation**
- [ ] Javadoc
- [ ] HAPI protobuf comments
- [ ] README

### CODE REVIEW

File: hapi-fees/src/main/java/com/hedera/services/usage/token/TokenMintUsage.java [hunk 1]
```diff
@@ -41,9 +45,26 @@ TokenMintUsage self() {
...
+	public TokenMintUsage givenSubType(SubType subType) {
+		this.currentSubType = subType;
+		return this;
+	}
+
 	public FeeData get() {
+		var op = this.op.getTokenMint();
+
+		// TODO: Split logic for calculations
+		if (currentSubType == SubType.TOKEN_NON_FUNGIBLE_UNIQUE) {
```
Comment (tinker-michaelj) [inline hapi-fees/src/main/java/com/hedera/services/usage/token/TokenMintUsage.java:58]: Can be done in a future PR, but we've noticed that the Streams API has too much overhead for use in the `handleTransaction` critical perf path.

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

File: hapi-fees/src/main/java/com/hedera/services/usage/token/TokenMintUsage.java (hunk 1 above)
Comment (georgiyazovaliiski) [reply hapi-fees/src/main/java/com/hedera/services/usage/token/TokenMintUsage.java:57]: I replaced the logic accordingly.

**Review by Daniel-K-Ivanov — APPROVED** [review]
LGTM! Lets add e2e tests to the logic in another PR

**Pull request merged**